daffodil-test/src/test/scala/org/apache/daffodil/section23/dfdl_expressions/TestBitFunctions.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.section23.dfdl_expressions
+
+import org.apache.daffodil.tdml.Runner
+import org.junit._
+
+
+object TestBitFunctions {
+  val testDir = "/org/apache/daffodil/section23/dfdl_functions/"
+
+  val runner = Runner(testDir, "BitFunctions.tdml")
+
+  @AfterClass def shutDown(): Unit = {
+    runner.reset
+  }
+}
+
+class TestBitFunctions {
+  import TestBitFunctions._
+  @Test def test_longShiftTest01():Unit = {runner.runOneTest("longShiftTest")}
+  @Test def test_intShiftTest02():Unit = {runner.runOneTest("intShiftTest")}
+  @Test def test_byteShiftTest03():Unit = {runner.runOneTest("byteShiftTest")}
+  @Test def test_shortShiftTest04():Unit = {runner.runOneTest("shortShiftTest")}
+
+  @Test def test_leftShift_int_06():Unit = {runner.runOneTest("leftShiftIntError02")}
+  @Test def test_leftShift_int_08():Unit = {runner.runOneTest("rightShiftIntError02")}
+
+  @Test def test_leftShift_int_10():Unit = {runner.runOneTest("rightShiftLongError02")}
+  @Test def test_leftShift_int_12():Unit = {runner.runOneTest("leftShiftLongError02")}
```

Comment:
Some of these function names don't match the test name, and sound like they're testing different things. The convention is that the function name should be the same as the parserTestCase name but with the ``test_`` prefix.

I'd also suggest making all the names more consistent. For example you have ``longShiftTest`` for long, but ``testUnsignedIntShifitng`` for unsigned long. Sometimes you have a numbera the end, sometime you don't, etc.

I'd also prefer to see all the positive tests first, then all the negative tests. Right now, all the unsigned tests are at the bottom. Makes it easier to see that we have everything coverered that way.